lib/npg_qc/mqc/outcomes.pm

```diff
@@ -35,48 +31,51 @@ has 'qc_schema' => (
 
 sub get {
   my ($self, $qlist) = @_;
+
   if (!$qlist || (ref $qlist ne 'ARRAY')) {
     croak q[Input is missing or is not an array];
   }
 
-  my $hashed_queries = {};
-  foreach my $q ( @{$qlist} ) {
-    if (!defined $q->{$IDRK} || !defined $q->{$PK}) {
-      croak qq[Both '$IDRK' and '$PK' keys should be defined];
-    }
-    push @{$hashed_queries->{$q->{$IDRK}}->{$q->{$PK}}},
-      defined $q->{$TIK} ? $q->{$TIK} : undef;
-  }
-
   my @lib_outcomes = ();
   my @seq_outcomes = ();
   my @uqc_outcomes = ();
 
-  foreach my $id_run ( keys %{$hashed_queries} ) {
+  my @mcompcompositions = ();
+  my $scomponents = {};
 
-    my @positions = keys %{$hashed_queries->{$id_run}};
-
-    foreach my $p ( @positions ) {
-      my @tags = @{$hashed_queries->{$id_run}->{$p}};
-      my $query = {$IDRK => $id_run, $PK => $p};
-      # Either get lib outcomes for a selection of tag indices or
-      # all lib outcomes for a lane.
-      if ( none { !defined } @tags ) {
-        $query->{$TIK} = \@tags;
-      }
-      push @lib_outcomes, $self->_create_query($LIB_RS_NAME, $QC_OUTCOME, $query)->all();
+  foreach my $rpt_list ( uniq @{$qlist} ) {
+    my $composition = _rpt_list2composition($rpt_list);
+    if ($composition->num_components == 1) {
+      my $c = $composition->get_component(0);
+      $scomponents->{$c->id_run()}->{$c->position()} = 1;
+    } else {
+      push @mcompcompositions, $composition;
     }
+  }
 
-    # Get seq outcomes for a lane.
-    my $q = {$IDRK => $id_run, $PK => \@positions};
-    push @seq_outcomes, $self->_create_query($SEQ_RS_NAME, $QC_OUTCOME, $q)->all();
+  # Get outcomes for single-component compositions
+  foreach my $id_run (keys %{$scomponents}) {
```

Comment:
I got lost for a while with this variable because I thought it was components instead of scomponents. maybe rename to single_components